garrysmod/lua/includes/modules/scripted_ents.lua

```diff
@@ -22,18 +22,35 @@ BaseClasses[ "filter" ] = "base_filter"
 local function TableInherit( t, base )
 
 	for k, v in pairs( base ) do
-
-		if ( t[k] == nil ) then
-			t[k] = v
-		elseif ( istable( t[k] ) ) then
-			TableInherit( t[k], v )
+		if ( t[ k ] == nil ) then
+			t[ k ] = v
+		elseif ( istable( t[ k ] ) && istable( v ) ) then
+			TableInherit( t[ k ], v )
 		end
-
 	end
 
-	t[ "BaseClass" ] = base
+end
+
+local function DeepCopy( t )
+
+	local new = {}
+	setmetatable( new, debug.getmetatable( t ) )

```

Comment:
The difference is that debug.getmetatable ignores the `__metatable` of the object. Since it's a copy method, I'd say we want to use the `debug` version.